File: python_code/df_to_csv.py
import pandas as pd
from js import Blob, document, window
# import pyodide

def df_to_csv(jsonStr):
    # data = pyodide.http.open_url(path)
    df = pd.read_json(path_or_buf=jsonStr, orient='split')
    with open('/test.csv', 'w') as f:
        df.to_csv(f, index=False, mode='w')
    blob = Blob.new([f], {type: 'text/csv'})
    url = window.URL.createObjectURL(blob)
    a = document.body.appendChild('a')
    a.setAttribute('download', 'test.csv')
    a.href = url
    a.click()
    window.URL.revokeObjectURL(url)
    document.body.removeChild(a)
    # window.location.assign(url)
    # return url

File: src/python_code_js_modules/df_to_csv.js
const df_to_csv = `

import pandas as pd
import json
from js import Blob, document, window

def df_to_csv(json_str):
    df = pd.read_json(path_or_buf=json_str, orient='split')
    with open('/test.csv', 'w') as f:
        df.to_csv(f, index=False, mode='w')
    with open('/test.csv', 'r') as f:
        # return f.read(), pd.DataFrame.from_dict(json.loads(json_str)).to_json(orient='records')
        return json.dumps({
            "csv": f.read(),
            "xlsx": df.to_json(orient='records')
        })`;

export default df_to_csv;